tests/test_plugin.py

```diff
@@ -1534,3 +1537,83 @@ def test_3847_repro(node_factory, bitcoind):
     # This call to paystatus would fail if the pay plugin crashed (it's
     # provided by the plugin)
     l1.rpc.paystatus(i1)
+
+
+def test_important_plugin(node_factory):
+    # Cache it here.
+    pluginsdir = os.path.join(os.path.dirname(__file__), "plugins")
+
+    # Check we fail if we cannot find the important plugin.
+    n = node_factory.get_node(options={"important-plugin": "nonexistent"},
+                              may_fail=True, expect_fail=True,
+                              allow_broken_log=True)
+    assert not n.daemon.running
+    assert n.daemon.is_in_stderr('An --important-plugin=nonexistent was not found')
+
+    # Check we exit if the important plugin dies.
+    n = node_factory.get_node(options={"important-plugin": "fail_by_itself.py",
+                                       "plugin": os.path.join(pluginsdir, "fail_by_itself.py")},
+                              may_fail=True, expect_fail=True,
+                              allow_broken_log=True)
+
+    n.daemon.wait_for_log('Plugin marked as important by --important-plugin=fail_by_itself.py')
+    wait_for(lambda: not n.daemon.running)
+
+    # Check if the important plugin is made unimportant,
+    # we go back to ignoring it.
+    n = node_factory.get_node(options=OrderedDict([("important-plugin", "fail_by_itself.py"),
+                                                   ("unimportant-plugin", "fail_by_itself.py"),
```

Comment:
is `unimportant-plugin` necessary? it seems like `disable` might work better. i.e. are there cases where you'd want to run an 'important' plugin but not having it be fatal? if not, it seems like you could do away with the `unimportant` option and use `disable` as the override.

Reply:
You could be developing `pay`, for example, and want to recompile `plugins/pay` without having to restart `lightningd`.  You do this by marking it `--unimportant-plugin` and then using the `lightning-cli plugin restart pay` command.

Might be too edge case however...

Reply:
one alternative would be to have `--plugin` do the same work as as `--unimportant-plugin`. in other words, to mark an important plugin as unimportant you'd just need to specify it as `--plugin`? that seems straightforward enough.

Reply:
Certainly an idea.  However, how do we implement `add-plugin-dir`? Currently this is just equivalent to iterating `--plugin` over all executables in the dir,   But one thing a user might want to do would be to `add-plugin-dir` a directory containing all plugins they have, then later, if the decide that a plugin in that dir is important, use `important-plugin` on the "blessed" plugin.  In the current version of the PR the `important-plugin` upgrades the plugin to important, and a later `--plugin` or `--add-plugin-dir` will not downgrade it to unimportant.  Hmmm...........

Reply:
Alternatively, we should note that only the built-in plugins get default-important.  And the only ones interested in making built-in plugins unimportant are we the developers of `lightningd` (and not, say, developers building on top of `lightningd`).  So maybe better would be a `--dev-builtin-plugins-unimportant` for `DEVELOPER=1` builds?